pkg/engine/plan.go

```diff
@@ -311,6 +311,17 @@ func (acts *planActions) OnResourceStepPost(ctx interface{},
 			op, record = step.(*deploy.RefreshStep).ResultOp(), true
 		}
 
+		if step.Op() == deploy.OpRead &&
+			step.Old().Outputs != nil &&
```

Comment:
`step.Old()` can be nil if there's no existing resource in the snapshot - can you do a nil check here?